scene/main/viewport.cpp

```diff
@@ -1184,56 +1182,67 @@ void Viewport::_gui_show_tooltip() {
 	}
 
 	// Get the Control under cursor and the relevant tooltip text, if any.
-	Control *tooltip_owner = nullptr;
+	Control *tooltip_serve_target = nullptr;
 	String tooltip_text = _gui_get_tooltip(
 			gui.tooltip_control,
 			gui.tooltip_control->get_global_transform().xform_inv(gui.last_mouse_pos),
-			&tooltip_owner);
+			&tooltip_serve_target);
 	tooltip_text = tooltip_text.strip_edges();
 	if (tooltip_text.is_empty()) {
 		return; // Nothing to show.
 	}
 
-	// Remove previous popup if we change something.
-	if (gui.tooltip_popup) {
-		memdelete(gui.tooltip_popup);
-		gui.tooltip_popup = nullptr;
-		gui.tooltip_label = nullptr;
+	// Remove previous custom tooltip if we change something.
+	if (gui.custom_tooltip) {
+		memdelete(gui.custom_tooltip);
```

Comment:
This should also be nulled just in case the function returns below.

Reply:
Yes, thanks for pointing it out.